Assets/Scripts/InventoryAndItemsScripts/InventoryChecker.cs

```diff
@@ -0,0 +1,118 @@
+using Items;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Inventory
+{
+    /// <summary>
+    /// Can check the PlayerInventory based on SerializedFields.
+    /// (Only fill out relevant SerializedFields--ones that are not edited will be ignored)
+    /// </summary>
+    public class InventoryChecker : MonoBehaviour
+    {
+        [Tooltip("ItemData for equipment that needs to be currently equipped")]
+        [SerializeField] ItemData mustHaveEquipped;
+
+        [Tooltip("ItemData for equipment that must NOT be currently equipped")]
+        [SerializeField] ItemData mustNotHaveEquipped;
+
+        [Tooltip("ItemData and minimum stack size for items player must have in inventory")]
+        [SerializeField] List<InventoryItem> mustHaveTheseItems;
+
+        //The following may be useful if the criteria may be met by multiple different items:
+
+        [Tooltip("Player must have at least one of the entries in this List of ItemData")]
+        [SerializeField] List<ItemData> mustHaveAtLeastOneOfTheseItems;
+
+        [Tooltip("Player must have currently equipped at least one of the entries in this List of ItemData")]
+        [SerializeField] List<ItemData> mustHaveEquippedAtLeastOneOfTheseItems;
+
+        /// <summary>
+        /// Checks the PlayerInventory.
+        /// Returns true if the PlayerInventory meets the criteria
+        /// in SerializedFields for this 
+        /// </summary>
+        public bool InventoryStateMeetsCriteria()
+        {
+            if (this.mustHaveEquipped != null &&
```

Comment:
!= null is fine in this case, in C# it is better practice to use <var> is not null for more expected behavior
https://stackoverflow.com/questions/69267179/what-is-the-difference-between-is-not-null-and-null
not a necessary change here but is a good practice to get into